bin/arnold

```diff
@@ -0,0 +1,1009 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+# Docker
+declare -r ARNOLD_IMAGE_NAME="${ARNOLD_IMAGE_NAME:-fundocker/arnold}"
+declare -r ARNOLD_IMAGE_TAG="${ARNOLD_IMAGE_TAG:-5.22.0}"
+declare -r ARNOLD_IMAGE="${ARNOLD_IMAGE:-${ARNOLD_IMAGE_NAME}:${ARNOLD_IMAGE_TAG}}"
+declare -r ARNOLD_PROJECT_PATH="${PWD}"
+declare -r ARNOLD_TRAY_FILE="arnold.yml"
+declare -r DOCKER_USER="${DOCKER_USER:-$(id -u)}"
+
+# Arnold release info
+declare LATEST_RELEASE
+declare LATEST_VERSION
+
+# Script infos
+declare SCRIPT_NAME
+SCRIPT_NAME=$(basename "$0")
+declare SCRIPT_PATH
+SCRIPT_PATH="$(dirname "$(readlink -f "$0")")/${SCRIPT_NAME}"
+
+# The team file lists emails corresponding to public gpg keys that will be used
+# to encrypt ansible vault password using gpg.
+declare -r TEAM_FILE="${ARNOLD_PROJECT_PATH}/.team"
+
+# Set ansible vault password default to the corresponding environment variable
+# value (insecure mode, while not using gpg or --ask-vault-pass argument in
+# ansible-playbook command calls).
+declare ANSIBLE_VAULT_PASSWORD="${ANSIBLE_VAULT_PASSWORD}"
+
+# Terminal colors
+declare -r COLOR_BLUE='\033[0;36m'
+declare -r COLOR_DEFAULT='\033[0;39m'
+declare -r COLOR_GREEN='\033[0;32m'
+declare -r COLOR_ORANGE='\033[0;33m'
+declare -r COLOR_RED='\033[0;31m'
+declare -r COLOR_RESET='\033[0m'
+
+# Log levels
+declare -i LOG_LEVEL_CRITICAL=50
+declare -i LOG_LEVEL_ERROR=40
+declare -i LOG_LEVEL_WARNING=30
+declare -i LOG_LEVEL_SUCCESS=25
+declare -i LOG_LEVEL_INFO=20
+declare -i LOG_LEVEL_DEBUG=10
+declare -i LOG_LEVEL_NOTSET=0
+declare -i DEFAULT_LOG_LEVEL=${LOG_LEVEL_INFO}
+declare -i LOG_LEVEL=${DEFAULT_LOG_LEVEL}
+declare -A LOG_LEVELS=(
+  ["critical"]=${LOG_LEVEL_CRITICAL}
+  ["error"]=${LOG_LEVEL_ERROR}
+  ["warning"]=${LOG_LEVEL_WARNING}
+  ["success"]=${LOG_LEVEL_SUCCESS}
+  ["info"]=${LOG_LEVEL_INFO}
+  ["debug"]=${LOG_LEVEL_DEBUG}
+  ["notset"]=${LOG_LEVEL_NOTSET}
+)
+declare -A LOG_LEVEL_COLORS=(
+  ["critical"]=${COLOR_RED}
+  ["error"]=${COLOR_RED}
+  ["warning"]=${COLOR_ORANGE}
+  ["success"]=${COLOR_GREEN}
+  ["info"]=${COLOR_BLUE}
+  ["debug"]=${COLOR_DEFAULT}
+  ["notset"]=${COLOR_DEFAULT}
+)
+
+# Directories
+declare -r RELATIVE_COMMON_DIR="group_vars/common"
+declare -r RELATIVE_CUSTOMER_DIR="group_vars/customer"
+declare -r COMMON_DIR="${ARNOLD_PROJECT_PATH}/${RELATIVE_COMMON_DIR}"
+declare -r CUSTOMER_DIR="${ARNOLD_PROJECT_PATH}/${RELATIVE_CUSTOMER_DIR}"
+
+# Files
+declare -r VAULT_PASSWORD_FILE_NAME="password.gpg"
+
+# Dependencies
+declare -r CURL="curl"
```

Comment:
This is an abstraction layer in case we decide to change our mind and bundle it with Arnold for example. I find it convenient to define all dependencies in the same code block. But of course, this can be changed.

WDYT @madmatah?

Reply:
Ok but at the end of the script you added a link to the curl website to install it. What will you do if you change the value of this variable ?

Reply:
I think it's a good practice and consistent with other tools used in this script.

BUT I have a problem with the way it's used line 971 to 975, that limit the possible values here.

Reply:
> BUT I have a problem with the way it's used line 971 to 975, that limit the possible values here.

True. Should I remove this check then? What do you suggest?

Reply:
removing the `CURL` variable ?

Reply:
Same here, I would remove the variable and the `eval` L263.

Reply:
Done both. Thanks!